### DISCUSSION TITLE
Great functionality to merge (multiple) databases.

### DISCUSSION THREAD

**Discussion by meteos77** [textbrowser/biblioteq#133] [General]
BRAVO you have done a SUPER SUPER job

I did a great job of joining my 2 bases.

My 3rd try - I had errors (the one with 7018 documents)
1 - six books were duplicated by isbn13 so normal
2 - six books were on loan, it gave me errors on memberid ...memberhitory

the main functionality is GREAT
you can populate the main database by adding little by little the shelves

the error messages are not yet in the error logs

**Comment by textbrowser** [comment]
It's incomplete because BIGINT UNIQUE fields have to be recomputed.

I'll try to replicate the member problems.

**Reply by textbrowser** [reply]
The export can include the images. I don't know if that's valuable.

**Reply by meteos77** [reply]
the option to keep the images can be a plus for personal libraries
for us has the library with more than 8000 books i think the base will become too huge if we add the covers. and so problem with the backups ...

**Comment by textbrowser** [comment]
Yah, myoid fields are annoying. I can update a myoid in book and SQLite will not preserve the item_oid value in book_copy_info.

**Reply by meteos77** [reply]
I didn't understand your way of handling copies from the beginning
for us the only unique number is the access number (one barcode created per document). If we have 2 identical documents (same isbn), they have a different access number. This allows us to manage the copy in a unique way (for example: 1 is new and the other is in an average condition or the different purchase price or the different storage finally all the information as a unique copy).

**Comment by textbrowser** [comment]
If you're merging databases, myoids will collide. So, each insert into the target database has to set the myoid = max(...) + 1. Then adjust the respective child tables to maintain the relationships.

**Reply by meteos77** [reply]
To copy a base I would be of the opinion that the documents should not be borrowed
the members of the copied database are not copied etc... 
otherwise I understand the number of difficulties to manage, renumbering members, copies, loans, collisions of isbn and I forget surely

for my request the copy function was i have documents in a base and i want to integrate them in the main base. so renumbering of the myoids of the imported data and management of the duplicates of isbn by reporting them in a log file.

After which relations do they remain if they are not borrowed?

**Comment by textbrowser** [comment]
The merge will insert two unique books (having distinct ISBNs) which have the same myoids. This is a problem.

**Reply by meteos77** [reply]
it's you who manage the machinery behind :-)

what I have trouble understanding is that I can't find 
neither the barcode field (isbn-1 isbn-2 ... for the copies), nor the myoid field (except in the wonderful "tool tips") but not in the table columns, nor in the .csv export

the total of the documents represents only one document even if we make 1 copy

well we have very very few duplicate documents, but it is especially if we import books from the departmental library (300 to 500 books at once) and that there is a duplicate with one of our own collection (the isbn will be the same)

**Comment by textbrowser** [comment]
Schema definition.

https://github.com/textbrowser/biblioteq/blob/master/Source/biblioteq_sqlite_create_schema.h

**Reply by meteos77** [reply]
it's a bit too much in the engine for me :-)
a small question by curiosity: 
why myoid is not defined in UNIQUE, I thought it was this field that really defined each document because the different document books, cd, dvd, newspapers ... do not all have isbn

**Comment by textbrowser** [comment]
It isn't necessary. For data integrity, I agree that it would help.

**Reply by meteos77** [reply]
I resonate with the current management of the library
the UNIQUE fields are myoid (internal management of the database) and accession number (human management of the books - we < 10000 books so access number has 4 digits easy to type on the keyboard) or barcode label with shower head
the library had decided at the time to use the myoid of the software as accession number on the physical book.

result after 20 years, we have a unique reference that we put in accession number but on the > 7000 documents we have only 1500 documents with ISBN.
not to mention the old books printed before 1970 that don't have ISBN and make the volunteers juggle with the grey literature... no no

for the new books I take your idea now it will be the printed isbn
for the old ones, I will take them back 1 by 1 to note the isbn
 for the < 1970 I will create fake ISBN. and create barcode labels for the scan.

Do you see a problem in my method?

**Comment by meteos77** [comment]
I come back to a previous request
the search on accession number should be in the general ID field as well as isbn-10 isbn-13 because almost all the documents in the database have this field in their table (books, cd, dvd, newspapers, magazine, video, photo) the only one is litterature grise which doesn't have it ?

people who are looking for a document may need it without going to the specific menu of each category

**Comment by textbrowser** [comment]
That will be a long change. New queries are tedious. :P

**Reply by meteos77** [reply]
as I really don't know anything about programming ; I thought it was easy :-)

you can't do something like a shortcut where it executes a user-defined query and have a field asking for just the number 7056
for me it will be the query :
SELECT * FROM book WHERE accession_number='7056

for others SELECT * FROM book WHERE lc_control_number=469546

leave free the choice of the query but in the first table

**Comment by textbrowser** [comment]
There are a bunch of item types. A union of queries has to be created. It doesn't magically appear.

**Reply by meteos77** [reply]
What is a union of queries?
you mean that SELECT * FROM book OR from dvd is not easy to do 
(I know you are a magician but behind every magician there is a huge work to do the magic trick)

**Reply by textbrowser** [reply]
The query is composed of multiple queries because there are multiple tables of independent data; or compound selects.

**Comment by textbrowser** [comment]
It has to work for PostgreSQL and SQLite.

**Reply by meteos77** [reply]
completely agree

**Comment by textbrowser** [comment]
Completed! :)

**Reply by textbrowser** [reply]
Accession Number query.

**Reply by meteos77** [reply]
400 lines of code for this feature 8-) Thanks to the wizard !
and cherry on the cake : accession_number is at the beginning of the list so 1 click less :-)

**Comment by textbrowser** [comment]
Accidental. :P
